extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftUtils.java

```diff
@@ -147,4 +147,8 @@ public static OpenshiftConfig mergeConfig(OpenshiftConfig openshiftConfig, S2iCo
 
         return result;
     }
+
+    private static boolean hasProperty(String name, Set<String> names) {
+        return names.contains(name) || System.getProperties().containsKey(name);
```

Comment:
What's the rationale behind this?

Reply:
We want to check if a property is present in the config object or is passed via system properties.

Reply:
Sure, but why do we need to check systsme properties? The Config object already contains a source for system properties

Reply:
This should not be the case... When the `Config` is created, some sources are added by default, like the `SysPropConfigSource`. If that is not happening here, we need to first understand why.

Reply:
My guess is that this is due to the fact that the config phase is BUILD_TIME.

Reply:
Hmmm, my previous comment doesn't make sense, as we are talking about BUILD_TIME.

Reply:
Using something like this `mvnDebug clean install -Dquarkus.s2i-base-jvm-image=test` does not seem to work.

Reply:
Let me elaborate. Debuging the build with a command like the one above results in the config not containing the property, even though the system properties source is listed.

It seems that the root cause is the System property not being visible at all.
I will close the PR.